src/components/Choice/README.md

```diff
@@ -0,0 +1,186 @@
+
+# Choice
+
+## Examples
+
+### Default
+
+```vue
+<template>
+	<div>
+		<m-choice v-model="selected">
+			<m-choice-option value="choice-1">
+				Choice
+			</m-choice-option>
+			<m-choice-option value="choice-2">
+				Choice
+			</m-choice-option>
+			<m-choice-option value="choice-3">
+				Choice
+			</m-choice-option>
+			<m-choice-option
+				:disabled="true"
+				value="choice-4"
+			>
+				Choice
+			</m-choice-option>
+		</m-choice>
+		<br>
+		<p>Selected value: {{ selected }}</p>
+	</div>
+</template>
+
+<script>
+import { MChoice, MChoiceOption } from '@square/maker/components/Choice';
+
+export default {
+	components: {
+		MChoice,
+		MChoiceOption,
+	},
+	data() {
+		return {
+			selected: 'choice-1',
+		};
+	},
+};
+</script>
+```
+
+### Multi select mode
+
+```vue
+<template>
+	<div>
+		<m-choice
+			v-model="selected"
+			mode="multi-select"
+		>
+			<m-choice-option value="choice-1">
+				Choice
+			</m-choice-option>
+			<m-choice-option value="choice-2">
+				Choice<br>second line
+			</m-choice-option>
+			<m-choice-option value="choice-3">
+				Choice longer text
+			</m-choice-option>
+			<m-choice-option
+				:disabled="true"
+				value="choice-4"
+			>
+				Choice disabled
+			</m-choice-option>
+		</m-choice>
+		<br>
+		<p>Selected value: {{ selected }}</p>
+	</div>
+</template>
+
+<script>
+import { MChoice, MChoiceOption } from '@square/maker/components/Choice';
+
+export default {
+	components: {
+		MChoice,
+		MChoiceOption,
+	},
+	data() {
+		return {
+			selected: ['choice-1', 'choice-4'],
+		};
+	},
+};
+</script>
+```
+
+### Fixed width
+
+```vue
+<template>
+	<div>
+		<m-choice v-model="selected">
+			<m-choice-option
+				value="choice-1"
+				class="choice-option"
+			>
+				Choice
+			</m-choice-option>
+			<m-choice-option
+				value="choice-2"
+				class="choice-option"
+			>
+				Choice longer text
+			</m-choice-option>
+			<m-choice-option
+				value="choice-3"
+				class="choice-option"
+			>
+				Choice
+			</m-choice-option>
+		</m-choice>
+		<br>
+		<p>Selected value: {{ selected }}</p>
+	</div>
+</template>
+
+<script>
+import { MChoice, MChoiceOption } from '@square/maker/components/Choice';
+
+export default {
+	components: {
+		MChoice,
+		MChoiceOption,
+	},
+	data() {
+		return {
+			selected: 'choice-1',
+		};
+	},
+};
+</script>
+
+<style scoped>
+.choice-option {
+	width: 172px;
```

Comment:
Shouldn't the text inside the component be responsible for setting the width instead of a static width via CSS?

Reply:
It does. But text can have uneven lengths and if the dev is going for a more uniform appearance they would have to set a fixed width themselves. `flexbox` doesn't work here because we'd have to set an explicit width on the parent container which we can't do here because `MChoice` is suppose to be allowed to have arbitrary dynamic widths while being thrown into a carousel-like component so that it's horizontally scrollable if it overflows.